models/user/userModel.js

```diff
@@ -0,0 +1,66 @@
+import mongoose, { Schema } from 'mongoose';
+import { addressSchema, orderItemSchema } from '../schemas';
+import { accountStatuses, roles } from '../../constants';
+import { phoneNumberPattern } from '../../helpers';
+
+
+const ObjectId = Schema.Types.ObjectId;
+
+const userSchema = new Schema(
+  {
+    fullName: {
+      type: String,
+      required: [true, 'Full name is required'],
+      minlength: [3, 'Full name must be at least 3 characters long'],
+    },
+    password: {
+      type: String,
+      required: [true, 'Password is required'],
+      minlength: [8, 'Password must be at least 8 characters long'],
+    },
+    email: {
+      type: String,
+      required: [true, 'Email is required'],
+      unique: true,
+      validate: {
+        validator: (value) => /\S+@\S+\.\S+/.test(value),
```

Comment:
can be put into validators file with `phoneNumber` and `ObjectId`

Reply:
Refactored